src/video/windows/SDL_windowsevents.c

```diff
@@ -1126,7 +1126,25 @@ LRESULT CALLBACK WIN_WindowProc(HWND hwnd, UINT msg, WPARAM wParam, LPARAM lPara
 
     case WM_MOUSEMOVE:
     {
-        /* SDL_Mouse *mouse = SDL_GetMouse(); */
+        SDL_Mouse *mouse = SDL_GetMouse();
+        int lparam_x_client = GET_X_LPARAM(lParam);
+        int lparam_y_client = GET_Y_LPARAM(lParam);
+
+        if (mouse->relative_mode) {         
```

Comment:
You probably want to check all of these states, since these are the conditions under which the cursor might be clipped:
```
    if ((mouse->relative_mode || (window->flags & SDL_WINDOW_MOUSE_GRABBED) ||
         (window->mouse_rect.w > 0 && window->mouse_rect.h > 0)) &&
        (window->flags & SDL_WINDOW_INPUT_FOCUS)) {
```